python/cudf/cudf/_lib/pylibcudf/io/types.pyx

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from libcpp.string cimport string
+from libcpp.utility cimport move
+from libcpp.vector cimport vector
+
+from cudf._lib.pylibcudf.libcudf.io.types cimport (
+    host_buffer,
+    source_info,
+    table_with_metadata,
+)
+
+import errno
+import io
+import os
+
+
+cdef class TableWithMetadata:
+    """A container holding a table and its associated metadata
+    (e.g. column names)
+
+    For details, see :cpp:class:`cudf::io::table_with_metadata`.
+    """
+
+    @property
+    def columns(self):
+        """
+        Return a list containing the columns of the table
+        """
+        return self.tbl._columns
+
+    @property
+    def column_names(self):
+        """
+        Return a list containing the column names of the table
+        """
+        # TODO: Handle nesting (columns with child columns)
```

Comment:
*nit*: perhaps raise if unhandled?

Reply:
I added an assert (since this hopefully should just be temporary until I get around to implementing a reader that can read child columns).